src/main/java/ch/njol/skript/entity/EntityData.java

```diff
@@ -259,7 +274,7 @@ public static <E extends Entity, T extends EntityData<E>> void register(
 
 	public EntityData() {
 		for (EntityDataInfo<?> info : infos) {
-			if (getClass() == info.getElementClass()) {
+			if (getClass().equals(info.getElementClass())) {
```

Comment:
what's the reason for this? i don't believe there is a difference.

Reply:
I believe this was changed when we were trying to find the the condition never passed and kept throwing the exception, it was an early on change.